packages/monitor-v2/src/monitor-polymarket/MonitorProposalsOrderBook.ts

```diff
@@ -50,11 +50,18 @@ export async function monitorTransactionsProposedOrderBook(
   const markets = await getPolymarketMarkets(params);
 
   const marketsWithAncillary = await getMarketsAncillary(params, markets);
+
   // Filter out markets that do not have a proposal event.
   const marketsWithEventData: PolymarketWithEventData[] = marketsWithAncillary
-    .filter((market) => proposalEvents.find((event) => event.ancillaryData === market.ancillaryData))
+    .filter((market) =>
+      proposalEvents.find(
+        (event) => event.ancillaryData === market.ancillaryData && event.timestamp === market.requestTimestamp
```

Comment:
Should we check the identifier, too? Just to avoid spoofing? Also, assuming someone is trying to fool the bot, if someone sent a request at the same time from a different address with identical details, what would happen? Should we validate the requesting address matches the expected value, too?

Reply:
Yes, good point!